checker.cpp

```diff
@@ -1,22 +1,65 @@
-#include <assert.h>
-#include <iostream>
+#include "checker.h"
+
 using namespace std;
 
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    cout << "Temperature out of range!\n";
-    return false;
-  } else if(soc < 20 || soc > 80) {
-    cout << "State of Charge out of range!\n";
-    return false;
-  } else if(chargeRate > 0.8) {
-    cout << "Charge Rate out of range!\n";
-    return false;
-  }
-  return true;
+void BatteryManagementSystem::check_abnormal_measure(const float input, const float low_limit, const float high_limit){
+    if(input < low_limit)
+    {
+        cout << "The input value " << input << " is lower than the accepted limit " << low_limit << endl; 
+    }
+    else if(input > high_limit)
+    {
+        cout << "The input value " << input << " is higher than the accepted limit " << high_limit << endl;
+    }
+}
+
+bool BatteryManagementSystem::checkTemperature(){
+    bool returnValue = true;
+
+    if(_temperature < LOW_TEMPERATURE || _temperature > HIGH_TEMPERATURE) {
```

Comment:
same check happens 2 times if the code enters check_abnormal_measure.
Try to see if this can be reduced